app/components/organization/AgentsCount.js

```diff
@@ -7,52 +9,61 @@ import { formatNumber } from '../../lib/number';
 class AgentsCount extends React.Component {
   static propTypes = {
     organization: React.PropTypes.shape({
-      id: React.PropTypes.string.isRequired,
       agents: React.PropTypes.shape({
         count: React.PropTypes.number.isRequired
       })
-    })
+    }),
+    relay: React.PropTypes.object.isRequired
   };
 
   state = {
     agentCount: this.props.organization.agents ? this.props.organization.agents.count : 0
   };
 
   componentDidMount() {
-    PusherStore.on("organization_stats:change", this.handleStoreChange);
+    PusherStore.on("organization_stats:change", this.handlePusherWebsocketEvent);
   }
 
   componentWillUnmount() {
-    PusherStore.off("organization_stats:change", this.handleStoreChange);
+    PusherStore.off("organization_stats:change", this.handlePusherWebsocketEvent);
   }
 
-  handleStoreChange = (payload) => {
-    this.setState({ agentCount: payload.agentsConnectedCount });
+  // Like in MyBuilds, we don't take the Pusher data for granted - we instead
+  // take it as a cue to update the data store backing the component.
+  handlePusherWebsocketEvent = (payload) => {
+    if (this.state.agentCount !== payload.agentsConnectedCount) {
+      this.props.relay.forceFetch();
```

Comment:
@ticky `component/throttle` looks perfect! "Supports leading and trailing invocation." is awesome as well.